scrapers/nd/bills.py

```diff
@@ -1,229 +1,174 @@
-from datetime import datetime
+import logging
+import re
+import requests
 import lxml.html
+import time
+import random
+import datetime as dt
 from openstates.scrape import Scraper, Bill
-from .actions import NDCategorizer
-import re
-from utils import LXMLMixin
-
+from spatula import HtmlListPage, HtmlPage, XPath, CSS
 
-class NDBillScraper(Scraper, LXMLMixin):
-    """
-    Scrapes available legislative information from the website of the North
-    Dakota legislature and stores it in the openstates  backend.
-    """
 
-    categorizer = NDCategorizer()
+def random_sleep(avg_secs):
```

Comment:
Let's document this function. _Why_ do we need random sleeps? Also worth asking, should this be a general function (in the global `utils` or even openstates-core) instead?

Reply:
Yes, I'll add a comment. The sleeps were an attempt to put less strain on the server after having several runs that resulted in server errors indicating it might have been overloaded with requests. The randomization of the frequency and duration of the sleeps was to seem less-automated. The site's [robots.txt](https://ndlegis.gov/robots.txt) does not explicitly address these issues.

I could write a version of this function that could be added to global utils that could have default parameters for the magnitude of duration and frequency, so that future scraper writers could adjust them to suit specific needs of other sites.

However, if there is functionality within spatula or openstates-core already for something like this, I will defer to that. This was my ad hoc solution.

Reply:
Also I meant to comment out the print statement within that function so that it can be optional (and to delete the `print(bill_id)` on line 41). Both of which I can include on the next commit.

Reply:
I see no _problem_ with you adding the function. Just making sure you're going to explain it. This is the sort of function that people come upon later and are very confused by.

Also, remember to _always_ document magic numbers (e.g. the 7, 14, 1, and 2 in this function).